src/solana-client.ts

```
@@ -0,0 +1,532 @@
+      [Buffer.from('task'), Buffer.from(taskId)],
+      this.programId
+    );
+  }
+
+  private getRewardVaultPDA(tokenMint: PublicKey): [PublicKey, number] {
+    return PublicKey.findProgramAddressSync(
+      [Buffer.from('reward_vault'), tokenMint.toBuffer()],
+      this.programId
+    );
+  }
+
+  // Initialize reward pool
+  async initializeRewardPool(
+    platformAuthority: Keypair,
+    platformFeePercentage: number = 10
+  ): Promise<string> {
+    const [rewardPoolPDA] = this.getRewardPoolPDA();
+
+    const data = Buffer.alloc(1 + 1);
+    data.writeUint8(RewardPoolInstruction.InitializeRewardPool, 0);
+    data.writeUint8(platformFeePercentage, 1);
+
+    const instruction = new TransactionInstruction({
+      keys: [
+        { pubkey: rewardPoolPDA, isSigner: false, isWritable: true },
+        { pubkey: platformAuthority.publicKey, isSigner: true, isWritable: true },
+        { pubkey: SystemProgram.programId, isSigner: false, isWritable: false },
+        { pubkey: SYSVAR_RENT_PUBKEY, isSigner: false, isWritable: false },
+      ],
+      programId: this.programId,
+      data,
+    });
+
+    const transaction = new Transaction().add(instruction);
+    transaction.recentBlockhash = (
+      await this.connection.getLatestBlockhash()
+    ).blockhash;
+    transaction.feePayer = platformAuthority.publicKey;
+
+    const signature = await sendAndConfirmTransaction(
+      this.connection,
+      transaction,
+      [platformAuthority]
+    );
+
+    return signature;
+  }
+
+  // Get reward pool data
+  async getRewardPool(): Promise<RewardPool | null> {
+    const [rewardPoolPDA] = this.getRewardPoolPDA();
+
+    try {
+      const accountInfo = await this.connection.getAccountInfo(rewardPoolPDA);
+      if (!accountInfo) {
+        return null;
+      }
+
+      const rewardPool = RewardPool.deserialize(accountInfo.data);
+      return rewardPool;
+    } catch (error) {
+      console.error('Failed to get reward pool:', error);
```

Some methods catch and log errors but silently return null. We might want to propagate meaningful errors back to the caller so the backend can handle them appropriately.

---

I propagated meaningful errors to caller